### PR TITLE
feat: add final evaluation

### PR DESCRIPTION
**Pull request by leseb**
8ff3780 feat: add final evaluation
113078c wip:final eval
fd52c08 update sdg generated line in knowledge-yaml
fc7067b fix typo on task generate
339ae90 fix: do not use backticks
5e74166 fix: add final details to push model to S3
a28d739 bulk: mt-bench eval, final eval and trained model push to S3
0885fa5 chore: remove helpers directory

commit 8ff3780e6994a8be2a9d09d2ea767240cbc979d8
Author: Sébastien Han <seb@redhat.com>
Date:   Thu Oct 10 17:39:25 2024 +0200

    feat: add final evaluation
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

commit 113078c738356d900e82ee1f5dd904ed335f6a5e
Author: Michael Clifford <mcliffor@redhat.com>
Date:   Thu Oct 10 20:04:04 2024 -0400

    wip:final eval
    
    Signed-off-by: Michael Clifford <mcliffor@redhat.com>

commit fd52c08c3deb874e9fd86cc204efa4be72a4fe04
Author: sallyom <somalley@redhat.com>
Date:   Fri Oct 11 13:21:38 2024 -0400

    update sdg generated line in knowledge-yaml
    
    Signed-off-by: sallyom <somalley@redhat.com>

commit fc7067b0228d99dcea2ad8effe63271e0c4ac897
Author: Sébastien Han <seb@redhat.com>
Date:   Mon Oct 14 09:40:50 2024 +0200

    fix typo on task generate
    
    It used to be `/data/generated`.
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

commit 339ae90a9a973dc3944a75c138da729d1651adbd
Author: Sébastien Han <seb@redhat.com>
Date:   Mon Oct 14 09:41:33 2024 +0200

    fix: do not use backticks
    
    The shell from the python executor will interpret the content of the
    backtick.
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

commit 5e74166e41a786021dbc58b01fe3b5808422d222
Author: Sébastien Han <seb@redhat.com>
Date:   Mon Oct 14 09:43:24 2024 +0200

    fix: add final details to push model to S3
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

commit a28d73980f3c0f2c3a79210a6bcf1b4c9c78b196
Author: Sébastien Han <seb@redhat.com>
Date:   Mon Oct 14 23:00:15 2024 +0200

    bulk: mt-bench eval, final eval and trained model push to S3
    
    - do not print final eval scores in logs
    - use the correct model location for final push
    - fix job/cr watch
    - add progress bar for download/upload
    - add a few globale variables for eval steps
    - handle the case where pods are removed but the job exists
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

commit 0885fa5c452b5fb061add63faa07ec2cf70b3b43
Author: Sébastien Han <seb@redhat.com>
Date:   Tue Oct 15 17:12:32 2024 +0200

    chore: remove helpers directory
    
    THe functions from the helper are embedded into the components
    themselves. So they are left unused now.
    
    Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

File: eval/final/components.py [hunk 1]
```diff
@@ -34,15 +34,97 @@ def run_final_eval_op(
     import os
 
     import torch
-    from helpers import (
-        VLLM_SERVER,
-        launch_vllm,
-        stop_vllm,
```
Comment (tumido) [inline eval/final/components.py:40]: I don't think these helpers are used anywhere else or are they? Would you like to delete them?

**Review by tumido — COMMENTED** [review]
Just a few nits. Haven't test run the code though. 🙂

File: eval/final/components.py (hunk 1 above)
Comment (leseb) [reply eval/final/components.py:40]: They are not used anymore indeed. I guess I was hesitant to remove them, but let's do it.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**